docs/API/metadata.md

```diff
@@ -7,9 +7,30 @@ authenticated user, while other metadata is maintained internally by the server
 and can't be changed. Authenticated users can also add any additional metadata
 that might be of use.
 
->__NOTE__: right now the ability to search and filter using metadata is
-limited, but our intent is to be able to use any defined metadata value both
-for searches and to filter the results of [datasets/list](V1/list.md).
+Dataset metadata is represented as a set of nested JSON objects. There are four
+distinct key namespaces. These can be addressed (read or changed) at any level
+of the hierarchy using a dotted name path, for example `dataset.resource_id`
+for a dataset's resource ID, or `global.environment.cluster.ip`. The keys are
+lowercase alphabetic, plus digits, hyphen, and underscore: so `global.u-7` is
+OK, but `global.Frank` isn't.
+
+The four namespaces are:
+* `dataset` provides inherent attributes of the dataset, including the full
+`metadata.log` as `dataset.metalog`. Most of these attributes cannot be changed
+after creation.
+* `server` provides server management state about a dataset. Most of these
+cannot be changed by the user. While many may not be directly meaningful to the
+user they are visible.
+* `global` provides user-controlled metadata which can only be modified by the
+owner of the dataset, but is visible to anyone with read access to the dataset.
+By convention, a client should use a unique second-level key to avoid
+conflicting paths. For example, the Pbench Dashboard uses `global.dashboard`.
+* `user` provides a metadata namespace that's private to a user: each user will
+see their own set of nested object structure and values, and these are not
+shareable. Even if you don't own a dataset you can set your own private `user`
+metadata to help you categorize that dataset and to find it again.
```

Comment:
Is it sufficiently obvious from context that each metadata value is associated with a specific dataset?  I would be inclined to make that explicit:
> `user` provides a given dataset with a private-to-each-user metadata namespace: each user will see their own keys and values, and these are not shareable. Even if you don't own a dataset you can set your own private `user` metadata associated with it to help you categorize that dataset and to find it again.